File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        seen = {0:1}
        count = 0
        cs = 0 
        for x in nums:
            cs+=x
            need = cs-k
            if need in seen:
                count += seen[need]
            seen[cs] = seen.get(cs,0) +1 
        return count
        

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> arr;
        arr[0] = 1;
        int count=0;
        int cs = 0;
        int need = 0;
        for(int i=0;i<nums.size();i++){
            cs += nums[i]; 
            need = cs - k;
            if(arr.contains(need)){
                count += arr[need];
            }
            arr[cs]++;
            
        }

        return count;
    }
};